common/djangoapps/third_party_auth/saml.py

```diff
@@ -93,3 +95,158 @@ def wrapped_method(*args, **kwargs):
     def _config(self):
         from .models import SAMLConfiguration
         return SAMLConfiguration.current(Site.objects.get_current(get_current_request()))
+
+
+class SapSuccessFactorsIdentityProvider(SAMLIdentityProvider):
```

Comment:
@haikuginger I thought that you wanted to put this new code in `edx-enterprise` rather than here?

Also, I don't like this approach of subclassing `SAMLIdentityProvider` and adding a new field called `identity_provider_type`, when we already have the existing functionality you need.

Instead of subclassing `SAMLIdentityProvider` I would rather see you subclass `SAMLAuthBackend`, which also has the same `get_user_details()` method that you need to override.

Then, you can use the existing `SAMLProviderConfig.backend_name` field to choose whether to use the SAP-specific backend or the standard `SAMLAuthBackend` (`tpa-saml`) backend. I think that would be simpler and better, reducing complexity and re-using existing patterns.